### PR TITLE
Update AMI terraform config.

### LINKED ISSUE #3111: Update foreman AMI
### Context
The current foreman image is quite out of date with a lot of pending updates 

### Problem or idea
The image has to be updated with the minor updates (note that major version upgrade is also available)

### Solution or next step
Generate a new AMI with the most recent updates.

### PR DESCRIPTION
**Pull request by arkid15r**
## Issue Number

Resolves #3111 

## Purpose/Implementation Notes

Update terraform config for foreman AMI generation.

## Types of changes

What types of changes does your code introduce?
- Bugfix (non-breaking change which fixes an issue)

### CODE REVIEW

File: ami/instance.tf [hunk 1]
```diff
@@ -26,10 +25,14 @@ resource "aws_instance" "ubuntu-ami-template-instance" {
...
   # instance. For more information see the definition of this resource.
-  user_data = data.template_file.ubuntu_instance_user_data.rendered
+  user_data = templatefile("ubuntu-instance-user-data.tpl.sh", {
+      docker_apt_key = data.local_file.docker_apt_key.content
+    }
+  )
 
-  subnet_id = data.aws_subnet.ccdl_dev_subnet.id
   associate_public_ip_address = true
+  key_name = "data-refinery-key-circleci-prod"
```
Comment (davidsmejia) [inline ami/instance.tf:34]:
```suggestion
  key_name = "data-refinery-key-${var.user}-${var.stage}"
```

Comment (davidsmejia) [reply ami/instance.tf:34]: Use templated values so there isn't collision for dev/stage/prod

**Review by davidsmejia — APPROVED** [review]
Can we change the target of this pr to `feature/automatic-accessions`

File: ami/instance.tf (hunk 1 above)
Comment (arkid15r) [reply ami/instance.tf:35]: Not sure if these variables are applicable in `ami` context. Keeping as is for now.

**Pull request merged**